tests/reftests/admin-cache.unix.test

```diff
@@ -0,0 +1,406 @@
+[sit.1] synchronised (file://${BASEDIR}/arch-sit.tgz)
+[amet.1] synchronised (file://${BASEDIR}/arch-amet.tgz)
+Done.
+### sh cache-analysi.sh
+ipsum : md5 : exists
+ipsum : sha256 : not found
+ipsum : sha512 : not found
+--
+dolor : md5 : exists, link to sha256
+dolor : sha256 : exists
+dolor : sha512 : not found
+--
+sit : md5 : not found
+sit : sha256 : exists
+sit : sha512 : not found
+--
+amet : md5 : exists, link to sha512
+amet : sha256 : exists, link to sha512
+amet : sha512 : exists
+### : add hashes
+### opam admin add-hashes sha512 --package ipsum
+### opam admin add-hashes sha512 --package dolor
+### opam admin add-hashes md5    --package sit
+[file://${BASEDIR}/arch-sit.tgz] found in cache
+### opam admin add-hashes sha512 --package sit
+### opam show --just-file ./packages/ipsum/ipsum.1/opam --field url.src,url.checksum | "${IP_MD5}" -> IP_MD5 | "${IP_SHA512}" -> IP_SHA512
+url.src:      file://${BASEDIR}/arch-ipsum.tgz
+url.checksum: md5=IP_MD5, sha512=IP_SHA512
+### opam show --just-file ./packages/dolor/dolor.1/opam --field url.src,url.checksum | "${DL_MD5}" -> DL_MD5 | "${DL_SHA256}" -> DL_SHA256 | "${DL_SHA512}" -> DL_SHA512
+url.src:      file://${BASEDIR}/arch-dolor.tgz
+url.checksum: md5=DL_MD5, sha256=DL_SHA256, sha512=DL_SHA512
+### opam show --just-file ./packages/sit/sit.1/opam --field url.src,url.checksum | "${ST_MD5}" -> ST_MD5 | "${ST_SHA256}" -> ST_SHA256 | "${ST_SHA512}" -> ST_SHA512
+url.src:      file://${BASEDIR}/arch-sit.tgz
+url.checksum: sha256=ST_SHA256, md5=ST_MD5, sha512=ST_SHA512
+### sh cache-analysi.sh
+ipsum : md5 : exists
+ipsum : sha256 : not found
+ipsum : sha512 : not found
+--
+dolor : md5 : exists, link to sha256
+dolor : sha256 : exists
+dolor : sha512 : not found
+--
+sit : md5 : not found
+sit : sha256 : exists
+sit : sha512 : not found
+--
+amet : md5 : exists, link to sha512
+amet : sha256 : exists, link to sha512
+amet : sha512 : exists
+### : cache
+### opam admin cache | unordered
+[dolor.1] synchronised (cached)
+[ipsum.1] synchronised (cached)
+[sit.1] synchronised (cached)
+Done.
+### sh cache-analysi.sh
+ipsum : md5 : exists
+ipsum : sha256 : not found
+ipsum : sha512 : exists, link to md5
+--
+dolor : md5 : exists, link to sha256
+dolor : sha256 : exists
+dolor : sha512 : exists, link to md5
+--
+sit : md5 : exists, link to sha256
+sit : sha256 : exists
+sit : sha512 : exists, link to sha256
+--
+amet : md5 : exists, link to sha512
+amet : sha256 : exists, link to sha512
+amet : sha512 : exists
+### ::::::::::::::::::
+### :V: corrupted md5:
+### ::::::::::::::::::
+### <packages/lorem/lorem.1/opam>
+opam-version: "2.0"
+### <hex1>
+4dc968ff0ee35c209572d4777b721587d36fa7b21bdc56b74a3dc0783e7b9518afbfa200a8284bf36e8e4b55b35f427593d849676da0d1555d8360fb5f07fea2
+### <hex2>
+4dc968ff0ee35c209572d4777b721587d36fa7b21bdc56b74a3dc0783e7b9518afbfa202a8284bf36e8e4b55b35f427593d849676da0d1d55d8360fb5f07fea2
+### xxd -r -p hex1 arch-lorem.tgz
+### xxd -r -p hex2 arch-evil.tgz
```

Comment:
for the md5 collision test, I used `xxd`, but there is maybe a simpler way?

Reply:
it looks fine by me and looks reasonably portable. If it has issues it should be fine just replacing it later